cmd/main.go

```diff
@@ -305,7 +304,7 @@ func setupNim(mgr manager.Manager, signalHandlerCtx context.Context, kubeClient
 	var err error
 
 	nimState := os.Getenv("NIM_STATE")
-	if !slices.Contains([]string{"removed", ""}, nimState) {
+	if nimState != "removed" {
```

Comment:
I guess it would be better to have hare `if nimState  == "managed"`
this way it doesn't matter if it is empty or disabled.

Reply:
To me, we need a default value for this and if there is `NIM_STATE` in the env, then it will overwrite the default value.

>Can you clarify what you mean @Jooho ?

nvm, it is `main.go`  so does not need any uni tests

Reply:
@spolti if its all the same, we would prefer to keep the logic as is.
First checking to see if its removed and implementing logic 
and if its not perform the necessary actions

Reply:
> To me, we need a default value for this and if there is `NIM_STATE` in the env, then it will overwrite the default value. User can use custom servingruntime that might not have NIM_STATE env so we should control this in the code.
> 

Updated to include default as requested @Jooho

Reply:
> I guess it would be better to have hare if nimState  == "managed"

Hey @spolti, I just wanted to clarify that the reason we opted for checking _removed_ is that, although we only expect _managed_/_removed_ for _nim-state_. The backing enum also supports _forced_ and _unmanaged_.

See [constants](https://github.com/openshift/api/blob/master/operator/v1/types.go#L35).
See [CRD](https://github.com/opendatahub-io/opendatahub-operator/blob/main/config/crd/bases/components.platform.opendatahub.io_modelcontrollers.yaml#L98).